### DISCUSSION TITLE
Is this project production ready?

### DISCUSSION THREAD

**Discussion by adrianvdh** [node-ts/bus#227] [General]
Hi there, I've been a fan of .NET's MassTransit project and I wanted to know if this project is production ready?

Should I consider switching over to .NET for my use case?

**Comment by adenhertog** [comment]
hi @Adrianvdh 

we use this package in production across several systems - some with dozens of worker services that scale out over sqs, and handle many millions of messages per day. 

it's certainly production ready from this perspective, and has been iterated over for the last 8+ years

however it doesn't have the more specialised features of MassTransit, so if you're looking for a library to build on I suggest you weigh up what features you need and what languages you want to support to make an accurate choice.

**Reply by adrianvdh** [reply]
@adenhertog Thank you for the quick reply, this sounds good!

Very nicely written library, do you think you'd ever build a NestJs wrapper for it?

**Comment by gtoselli** [comment]
Hi @Adrianvdh it's depends on the context in which you want to use it.

If you want to use this lib for business-relevant workflows or communication between services the missing of transactional outbox pattern make it unsuitable.

**Comment by adrianvdh** [comment]
I've been building out a project using NestJs and following the hexagonal architecture with pretty much custom DDD primitives, custom DB transaction management, and Telemetry, etc... The works! (I know node-ts here has its own package, but I wanted some custom abstractions). Recently, I've been following a couple YT channels on .NET, and .NET 8 makes it seem so easy to do everything out of the box for you, where I've been spending months to get right on NodeJs.

For example each module housing its own TypeORM migrations was a lot of effort to get right. This stuff is pretty easy in .NET or Django, etc. I love NodeJs and its ease of use to get started, but I'm really starting to doubt the ecosystem's maturity for enterprise tooling. I'm not hating, I think node-ts/bus is a really powerful tool and an awesome effort by @adenhertog, really well done to bring over some of the enterprise patterns to the NodeJs ecosystem 🥳!

I guess I'm just a bit frustrated, spending months and months building my own internal framework where .NET does almost all of it out of the box. I wish there was a Spring/.NET robust version for NodeJs. NestJs tries to do this, but it can't compare to the power and maturity of Spring/.NET.

Sorry for the rant, thank you for answering my questions 🙏🏻
